### ISSUE TITLE
Contacts import not working properly

### ISSUE THREAD

**Issue by deimjons** [nextcloud/contacts#3705]
### Describe the bug

Today I tried to migrate contacts from iCloud to nextcloud. So biggest part of contacts just emty. I tried to investigate problem, and looks like nextcloud do not import not standart fields. For example WhatsApp. 

### Steps to reproduce

1. Go to contacts
2. Click on import contacts

### Expected behavior

Need to create not standard contact field during import or at least show to used dialog with a proposition to change the field name or you lose data from this field. In the current application, you can lose a very large amount of information if you decide to abandon corporate services in favor of self-hosted solution.

Very simple scenario. I decided to self-host my data and close all accounts in big companys services. Before clearing all data on services I decided to make an export of contacts and import them to Nextcloud. So after import contacts looks like imported successfully. The quantity of contacts is the same. So I decided to remove .vcf file because now all contacts are in the nextcloud. In a few weeks when I decide to call some person it will be so bad surprise for me that his contact card has just a name and a field with the number is empty. (because the number was saved in the field WhatsApp or another non-standard field). Thats it, the connection with the person is lost. Because the account in a social network is closed and prunged and Nextcloud imports contacts incorrectly. (( I think poper import and export it is basic functionality and this issue is CRITICAL!!!

### Actual behavior

It just not show any warning and no import data from that fields what can't recognize.

### Contact version

5.4.2

### Operating system

Docker on arm mac

### PHP engine version

None

### Web server

None

### Database

None

**Comment by deimjons** [comment]
Update. I use custom fields to write the name of the service provider for example. In some countries, you can't call free from one provider to another and people have few numbers. I don't want to call all of them mobile. 

I found a topic where people have disputed what type of contact fields should be present. I do not understand what the problem is to make a standard list of 2-3 fields and just add others when needed. For example when a person doing the import. And need to add a screen in the settings of the calendar where you can edit this list of custom fields. (so the set of fields will be different for different calendars). Also if you rename the field it should apply to all cards with this field. If you remove the field should be a dialog that asks you to choose the new field for data from the removed field or approve that you will lose data. If the user wants to lose data he should write the field name of what he removes, without this button will be blocked (like in git when you remove repo)

**Comment by rahilarious** [comment]
It is a [bug in ](https://github.com/nextcloud/contacts/issues/3503) firefox. Using chromium based browser fixes it. Alternative solution is to upload VCF file and then clicking on it will start importing

**Comment by marco74** [comment]
I have the same Issue in chromium on Debian. It also doesn't matter if I upload the file and click on it or import it using the contacts importer function.

What I found out so far: Obviously the vcf file is downloaded to the browsers runtime environment. There it's parsed and each single contact is uploaded to the server again using a PUT request. It seems that there is not difference between the two different methods. But Sometimes I got the hint that the connection to the server was lost. Immediately the number off errors while importing increased.
![image](https://github.com/nextcloud/contacts/assets/3092437/114ed3a2-1c7f-42ec-925e-1acf4bc35189)
Looking at the error console I saw this here:
![image](https://github.com/nextcloud/contacts/assets/3092437/a545dcc3-9ada-4a0b-b08c-e3704c99e07c)

I used chunks of 100 contacts in each vcf. Using CardDav I was able to check the amount of uploaded records. Sometimes 100 were uploaded sometimes just 99. So please catch the Server outage problem and try to resend the contacts.

Just to know: I just upgraded to NC 28.0.1

**Comment by marco74** [comment]
Just to add: I ended up importing the vcf into the KDE application Kontact as an offline address book. Then I also imported the Nextcloud address book. Now I could easily select all Contacts from the offline one to the Nextcloud address book. Finally I right clicked on the Nextcloud address book and syncronized it by using the update function in the context menu.

The enables other features as well. I had massive problems with some VCard Features using the default functions. E.g. embedded images led to errors and also some encoded stuff. Using the Kontact applikation I was able to go around the Webinterface from Nextcloud which seems to be responsible for some parsing errors.
